feat(curriculum): add prototype for space mission roster workshop

LGH831 opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

<!-- Feel free to add any additional description of changes below this line -->
This is part of Naomi's sprints for the JavaScript/Full-Stack certificate.

jdwilkin4 commented on fullstack-cert/js-projects/space-mission-roster/user-stories.md:
```diff
@@ -0,0 +1,11 @@
+1. You should create a `crew` array that stores astronaut objects with `id`, `name`, `role`, `evaEligible`, and `priority`.
```
this file can be removed since you are working on a workshop and not a lab.
we only add user stories for labs. not workshops. 

the user stories in the case were more for you to build your prototype. but they don't need to be added here. 👍🏾

jdwilkin4 commented on fullstack-cert/js-projects/space-mission-roster/script.js:
```diff
@@ -0,0 +1,81 @@
+// function that pushes a new astronaut, enforces unique id's, and logs confirmation
+function addCrewMember(crew, astronaut) {
+  // checking for duplicate IDs
+  const existingId = crew.some(member => member.id === astronaut.id);
```
higher order functions haven't been taught yet.

LGH831 commented:
Thank you for the guidance. After confirming with Naomi, I have removed higher order functions and replaced them with more rudimentary manual functions aligned with the earlier lessons in JS (up to JS fundamentals sub block). The most advanced technique in my prototype now would likely be the nested loops (bubble sort). If needed, I'd be glad to change anything in the current prototype. Thanks!

LGH831 commented:
Prototype has been improved with Jessica's suggestions. Another question came up regarding wording, specifically on 'EVA' vs 'Eva' for the function  `getEVAReadyCrew`. I named it as such per the user stories, but in your view(s), should it be renamed to `getEvaReadyCrew`? It'd make the prototype more consistent, I think.

jdwilkin4 reviewed (commented):
It looks like it should be EVA and not eva
https://en.wikipedia.org/wiki/Extravehicular_activity

So I would update all occurrences of `isEvaEligible` accordingly.

jdwilkin4 reviewed (approved):
This looks good to go in. 

If you are new to working on workshops, please read through the docs first.
https://contribute.freecodecamp.org/how-to-work-on-workshops/

It would also be good to go through one of the workshops to understand the language and flow. 

Make sure this goes in the correct order and module for the js v9 certification.
